pkg/kubectl/cmd/autoscale.go

```diff
@@ -118,62 +110,73 @@ func RunAutoscale(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []
 		return cmdutil.UsageError(cmd, fmt.Sprintf("generator %q not found.", generatorName))
 	}
 	names := generator.ParamNames()
-	params := kubectl.MakeParams(cmd, names)
-	name := info.Name
-	params["default-name"] = name
 
-	params["scaleRef-kind"] = mapping.GroupVersionKind.Kind
-	params["scaleRef-name"] = name
-	params["scaleRef-apiVersion"] = mapping.GroupVersionKind.GroupVersion().String()
+	for i := range infos {
+		info := infos[i]
+		mapping := info.ResourceMapping()
+		if err := f.CanBeAutoscaled(mapping.GroupVersionKind.GroupKind()); err != nil {
+			return err
+		}
 
-	if err = kubectl.ValidateParams(names, params); err != nil {
-		return err
-	}
-	// Check for invalid flags used against the present generator.
-	if err := kubectl.EnsureFlagsValid(cmd, generators, generatorName); err != nil {
-		return err
-	}
+		name := info.Name
+		params := kubectl.MakeParams(cmd, names)
+		params["default-name"] = name
 
-	// Generate new object
-	object, err := generator.Generate(params)
-	if err != nil {
-		return err
-	}
+		params["scaleRef-kind"] = mapping.GroupVersionKind.Kind
+		params["scaleRef-name"] = name
+		params["scaleRef-apiVersion"] = mapping.GroupVersionKind.GroupVersion().String()
 
-	resourceMapper := &resource.Mapper{
-		ObjectTyper:  typer,
-		RESTMapper:   mapper,
-		ClientMapper: resource.ClientMapperFunc(f.ClientForMapping),
-		Decoder:      f.Decoder(true),
-	}
-	hpa, err := resourceMapper.InfoForObject(object, nil)
-	if err != nil {
-		return err
-	}
-	if cmdutil.ShouldRecord(cmd, hpa) {
-		if err := cmdutil.RecordChangeCause(hpa.Object, f.Command()); err != nil {
+		if err = kubectl.ValidateParams(names, params); err != nil {
+			return err
+		}
+		// Check for invalid flags used against the present generator.
+		if err := kubectl.EnsureFlagsValid(cmd, generators, generatorName); err != nil {
 			return err
 		}
-		object = hpa.Object
-	}
-	// TODO: extract this flag to a central location, when such a location exists.
-	if cmdutil.GetFlagBool(cmd, "dry-run") {
-		return f.PrintObject(cmd, mapper, object, out)
-	}
 
-	if err := kubectl.CreateOrUpdateAnnotation(cmdutil.GetFlagBool(cmd, cmdutil.ApplyAnnotationsFlag), hpa, f.JSONEncoder()); err != nil {
-		return err
-	}
+		// Generate new object
+		object, err := generator.Generate(params)
+		if err != nil {
+			return err
```

Comment:
This bit of code was updated in my latest push. I use the `Visit()` to not only traverse the files but because it has error aggregation built in: https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/resource/visitor.go#L202